src/Nuvem.PharmacyManagementSystem.Pharmacy.Data/Data/IPharmacyDbContext.cs

```diff
@@ -0,0 +1,11 @@
+using Microsoft.EntityFrameworkCore;
+
+namespace Nuvem.PharmacyManagementSystem.Pharmacy.Data.Data;
+public interface IPharmacyDbContext : IDisposable
```

Comment:
Interesting approach. It is slightly different than what I had in mind - but i can see it may possibly get the job done.